File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/common/commandhandler/CommandHandler.java
package cloud.ambar.common.commandhandler;

import cloud.ambar.common.eventstore.PostgresTransactionalEventStore;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
abstract public class CommandHandler {
    final protected PostgresTransactionalEventStore postgresTransactionalEventStore;
    public abstract void handleCommand(Command command);
}

File: application/backend-credit-card-enrollment/backend-c#/Common/Command/CommandHandler.cs
using CreditCardEnrollment.Common.EventStore;

namespace CreditCardEnrollment.Common.Command;

public abstract class CommandHandler {
    protected readonly PostgresTransactionalEventStore _postgresTransactionalEventStore;

    protected CommandHandler(PostgresTransactionalEventStore postgresTransactionalEventStore) {
        _postgresTransactionalEventStore = postgresTransactionalEventStore;
    }

    public abstract void HandleCommand(Command command);
}